flocker/control/_protocol.py

```diff
@@ -90,10 +98,30 @@ class NodeStateCommand(Command):
     Used by a convergence agent to update the control service about the
     status of a particular node.
     """
-    arguments = [('node_state', NodeStateArgument())]
+    arguments = [('node_state', NodeStateArgument()),
+                 ('eliot_context', Unicode())]
     response = []
 
 
+def with_eliot_context(function):
+    """
+    Decorator for responders who accept an ``eliot_context`` argument
+    that deserializes the given Eliot context and runs the wrapped
+    function using that as the Eliot context.
+
+    No support for returned ``Deferreds`` at this point.
```

Comment:
I suggest making this a noisy failed.  If a `Deferred` is returned, raise an exception.